server/graphql/v2/query/collection/ExpensesCollectionQuery.ts

```diff
@@ -53,10 +54,16 @@ const updateFilterConditionsForReadyToPay = async (where, include, host): Promis
   }
 
   if (checkTaxForms) {
-    const expensesSubjectToTaxForm = expenses.filter(e => !TAX_FORM_IGNORED_EXPENSE_TYPES.includes(e.type));
+    const expensesSubjectToTaxForm = expenses.filter(expenseMightBeSubjectToTaxForm);
     if (expensesSubjectToTaxForm.length > 0) {
-      const expensesIdsSubjectToTaxForm = expensesSubjectToTaxForm.map(expense => expense.id);
-      expensesIdsPendingTaxForms = await queries.getTaxFormsRequiredForExpenses(expensesIdsSubjectToTaxForm);
+      const expenseIds = expensesSubjectToTaxForm.map(expense => expense.id);
+      const requiredLegalDocs = await loaders.Expense.taxFormRequiredBeforePayment.loadMany(expenseIds);
+      for (let i = 0; i < requiredLegalDocs.length; i++) {
+        if (requiredLegalDocs[i]) {
+          expensesIdsPendingTaxForms.add(expenseIds[i]);
+        }
+      }
+
```

Comment:
Modified this code to use the loader here as well. The main benefit is to preload the tax forms status in the loader, so the subsequent calls (in `Expense.requiredLegalDocuments` + `Expense.securityChecks`) directly get the result from the loader's cache.

Confirmed with a log in the SQL query function to make sure it's only called one time:

![image](https://github.com/opencollective/opencollective-api/assets/1556356/c79347a3-07d9-464a-ace0-ecc9a55cd600)